jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/ImageName.java

```diff
@@ -80,16 +83,30 @@ public ImageName(String fullName, String givenTag) {
 
         // set digest to null as default
         digest = null;
+
+        String algo_reg = "@sha(\\d+):";
+
+        Pattern pattern = Pattern.compile(algo_reg);
+
+        // Create a Matcher object
+        Matcher matcher = pattern.matcher(fullName);
+        // Set Hash length
+        if (matcher.find()) {
+            String numberString = matcher.group(1);
+            hashLen = Integer.parseInt(numberString);
+        }
+        Pattern digestPattern = Pattern.compile("@sha(?:[1-9][0-9]*|0b[01]+|0x[0-9a-fA-F]+)?");
```

Comment:
Have you tested this regexp? Or could you please point to it's source. This doesn't look right to me. Why are you declaring ranges `[1-9]` and `[0-9]` together? 

Shouldn't `[0-9a-fA-F]` be just `[0-9a-f]`?

Reply:
Yes, I have tested it.
For ranging from [1-9] and [0-9]: [1-9] is for first digit which may be from 1-9 and the 2nd one ([0-9]).
I will fix this part.